File: py101/easy3/9.py
# https://launchschool.com/exercises/f2e7ee97

# def clean_up(string):
#     new_string = []

#     for char in string:
#         if char.isalpha():
#             new_string.append(char)
#         elif not new_string and not char.isalpha():
#             new_string.append(" ")
#         elif not char.isalpha() and new_string[-1] == " ":
#             continue
#         elif not char.isalpha() and new_string[-1] != " ":
#             new_string.append(" ")

#     return "".join(new_string)

# print(clean_up("---what's my +*& line?") == " what s my line ")
# # True

def clean_up(string):
    new_string = []

    for char in string:
        if char.isalpha():
            new_string.append(char)
        elif not new_string or new_string[-1] != " ":
            new_string.append(" ")

    return "".join(new_string)

print(clean_up("---what's my +*& line?") == " what s my line ")
# True

File: js210/simpleProblems/easy2/9.js
// https://launchschool.com/exercises/1a7f4ec6

'use strict';

// const cleanUp = (string) => {
//   return string.replace(/\W+/g,' ');
// };

function isNotAlpha(char) {
  return (((char < 'a' || char > 'z') && ((char < 'A' || char > 'Z'))));
}

const cleanUp = (string) => {
  let result = '';

  for (let char of string) {
    if (isNotAlpha(char) && result[result.length - 1] === ' ') {
      continue;
    } else if (isNotAlpha(char) && result[result.length - 1] !== ' ') {
      result += ' ';
    } else {
      result += char;
    }
  }

  return result;
};

console.log(cleanUp("---what's my +*& line?"));    // " what s my line "